### PR TITLE
async with stack trace

### PR DESCRIPTION
**Pull request by Goxiaoy**
Add statck trace to async snapshot

### CODE REVIEW

File: packages/flutter_hooks/lib/src/async.dart [hunk 1]
```diff
@@ -83,11 +83,11 @@ class _FutureStateHook<T> extends HookState<AsyncSnapshot<T>, _FutureHook<T>> {
             _snapshot = AsyncSnapshot<T>.withData(ConnectionState.done, data);
           });
         }
-      }, onError: (dynamic error) {
+      }, onError: (dynamic error, dynamic stackTrace) {
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/async.dart:86]:
```suggestion
      }, onError: (dynamic error,  StackTrace stackTrace) {
```

File: packages/flutter_hooks/lib/src/async.dart [hunk 2]
```diff
@@ -186,9 +186,9 @@ class _StreamHookState<T> extends HookState<AsyncSnapshot<T>, _StreamHook<T>> {
         setState(() {
           _summary = afterData(data);
         });
-      }, onError: (dynamic error) {
+      }, onError: (dynamic error, dynamic stackTrace) {
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/async.dart:189]:
```suggestion
      }, onError: (dynamic error, StackTrace stackTrace) {
```

File: packages/flutter_hooks/lib/src/async.dart [hunk 3]
```diff
@@ -83,11 +83,11 @@ class _FutureStateHook<T> extends HookState<AsyncSnapshot<T>, _FutureHook<T>> {
...
           });
         }
-      }, onError: (dynamic error) {
+      }, onError: (dynamic error, dynamic stackTrace) {
         if (_activeCallbackIdentity == callbackIdentity) {
           setState(() {
-            _snapshot = AsyncSnapshot<T>.withError(
-                ConnectionState.done, error as Object);
+            _snapshot = AsyncSnapshot<T>.withError(ConnectionState.done,
+                error as Object, stackTrace as StackTrace);
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/async.dart:90]:
```suggestion
                error as Object, stackTrace);
```

File: packages/flutter_hooks/lib/src/async.dart [hunk 4]
```diff
@@ -186,9 +186,9 @@ class _StreamHookState<T> extends HookState<AsyncSnapshot<T>, _StreamHook<T>> {
         setState(() {
           _summary = afterData(data);
         });
-      }, onError: (dynamic error) {
+      }, onError: (dynamic error, dynamic stackTrace) {
         setState(() {
-          _summary = afterError(error as Object);
+          _summary = afterError(error as Object, stackTrace as StackTrace);
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/async.dart:191]:
```suggestion
          _summary = afterError(error as Object, stackTrace);
```

**Comment by rrousselGit** [comment]
Thanks!

Could you add tests?

**Comment by Goxiaoy** [comment]
```
Avoid annotating types for function expression parameters.dart(avoid_types_on_closure_parameters)
```

`onError: (dynamic error, StackTrace stackTrace) ` disobey the `avoid_types_on_closure_parameters` lint rule, should I disable that temporally?


And once `dynamic` or `StackTrace` removed, `implicit_dynamic_parameter` is triggered
```
Try adding an explicit type, or remove implicit-dynamic from your analysis options file.dart(implicit_dynamic_parameter)
```

**Pull request merged**